src/vm/dllimportcallback.cpp

```diff
@@ -1412,6 +1412,8 @@ VOID UMThunkMarshInfo::RunTimeInit()
         new (&sigInfo) PInvokeStaticSigInfo(pMD);
     else
         new (&sigInfo) PInvokeStaticSigInfo(GetSignature(), GetModule());
+    if (pStubMD == NULL)
+        m_cbActualArgSize += numRegistersUsed * STACK_ELEM_SIZE;
     if (sigInfo.GetCallConv() == pmCallConvCdecl)
```

Comment:
If we apply adjustment to `m_cbRetPop` only, then test failed with incorrect result. I suppose it failed due to [this check](https://github.com/dotnet/coreclr/blob/master/src/vm/i386/umthunkstub.S#L89). Probably, we should change `UMThunkStub` too?